p2p/debug/list.go

```diff
@@ -0,0 +1,130 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
+	"github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/p2p/utils"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+var peerListTableHeader = []string{"Peer ID", "Pokt Address", "ServiceURL"}
+
+// PrintPeerList retrieves the correct peer list using the peerstore provider
+// on the bus and then passes this list to PrintPeerListTable to print the
+// list of peers to os.Stdout as a table
+func PrintPeerList(bus modules.Bus, routerType RouterType) error {
+	var (
+		peers           types.PeerList
+		routerPlurality = ""
+	)
+
+	// TECHDEBT(#811): use `bus.GetPeerstoreProvider()` after peerstore provider
+	// is retrievable as a proper submodule.
+	pstoreProvider, err := peerstore_provider.GetPeerstoreProvider(bus)
+	if err != nil {
+		return err
+	}
+
+	switch routerType {
+	case StakedRouterType:
+		// TODO_IN_THIS_COMMIT: what about unstaked peers actors?
```

Comment:
@Olshansk Am I missing something here but what does `what about unstaked peers actors` mean?

When using the staked router type we are only looking for staked actors? If we want both we would use `AllRouterTypes`

Reply:
This TODO goes along with the commented code line below. It's referring to the current actor (this local P2P module instance ). My thinking at the time was something along the lines of: if the current actor is unstaked, it may not make sense to ask the peerstore provider for the staked actor set. Perhaps it should indeed work as written and it would be the responsibility of the P2P module to ensure the appropriate peerstore provider implemented is registered and then that of the node to ensure a compatible persistence module implementation is also registered (which may not yet exist), for example in the case of light client actors.

Reply:
@bryanchriswhite I appreciate your response during your time away so no rush on this reply. But AFAIK if we are checking whether the node is staked we need to retrieve the staked peerstore anyway? the `p2pModule.isStakedActor()` function is as follows:

```go
func (m *p2pModule) getStakedPeerstore() (typesP2P.Peerstore, error) {
	pstoreProvider, err := peerstore_provider.GetPeerstoreProvider(m.GetBus())
	if err != nil {
		return nil, err
	}

	return pstoreProvider.GetStakedPeerstoreAtHeight(
		m.GetBus().GetCurrentHeightProvider().CurrentHeight(),
	)
}

// isStakedActor returns whether the current node is a staked actor at the current height.
// Return an error if a peerstore can't be provided.
func (m *p2pModule) isStakedActor() (bool, error) {
	pstore, err := m.getStakedPeerstore()
	if err != nil {
		return false, fmt.Errorf("getting staked peerstore: %w", err)
	}

	// Ensure self address is present in current height's staked actor set.
	if self := pstore.GetPeer(m.address); self != nil {
		return true, nil
	}
	return false, nil
}
```

So if we check the addr from `bus.GetP2PModule().GetAddress()` against this staked list and its not there we return the unstaked peerstore? I feel like this defeats the purpose of us passing in the router type in the first place if we are going to ignore it? Or am I misunderstanding?